core/common/src/main/java/alluxio/collections/ConcurrentHashSet.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.collections;
+
+import java.util.AbstractSet;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A concurrent hash set. This is backed by a {@link ConcurrentHashMap}, and {@link Set} operations
+ * are translated to {@link ConcurrentHashMap} operations.
+ * @param <T> the type of the set objects
+ */
+@ThreadSafe
+public class ConcurrentHashSet<T> extends AbstractSet<T> {
```

Comment:
What do you think about using `Collections.newSetFromMap(new ConcurrentHashMap())`?